src/App.tsx

```diff
@@ -1,26 +1,218 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  creatTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+import { Header } from './conponents/Header/Header';
+import { TodoList } from './conponents/TodoList/TodoList';
+import { Footer } from './conponents/Footer/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<Status>(Status.All);
+  const [error, setError] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [newTitle, setNewTitle] = useState<string>('');
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setError(Error.UnableLoad));
+  }, []);
+
+  useEffect(() => {
+    if (!error) {
+      return;
+    }
+
+    const timeout = setTimeout(() => setError(''), 3000);
+
+    return () => clearTimeout(timeout);
+  }, [error]);
```

Comment:
[no need to fix]
technically you have a bug with this - if two errors happen within 3000 ms, the second error will be displayed only for a fraction of the time.
To avoid this you should clear (or check and clear)  the timeout before setting the new error

But this is very unlikely in your app, so no need to fix it